transform/mattermost-analytics/models/intermediate/product/active_users/int_user_active_days_legacy_telemetry.sql

```diff
@@ -12,24 +12,27 @@
 with tmp as (
     select
         cast(received_at as date) as received_at_date,
-        cast(timestamp as date) as activity_date,
+        cast(m2t.timestamp as date) as activity_date,
         server_id,
-        user_id
+        user_id,
+        client_type
     from
-        {{ ref('stg_mattermost2__tracks') }}
+        {{ ref('stg_mattermost2__tracks') }} m2t 
+    left join {{ ref('int_user_agent_parser') }} uap 
+        on m2t.context_user_agent = uap.context_user_agent
     where
         -- Exclude items without user ids, such as server side telemetry etc
         user_id is not null
         -- Exclude items without server ids
         and server_id is not null
         -- Exclude items with missing timestamps
-        and timestamp is not null
+        and m2t.timestamp is not null
         and received_at <= current_timestamp
 {% if is_incremental() %}
         -- this filter will only be applied on an incremental run
         and received_at >= (select max(received_at_date) from {{ this }})
 {% endif %}
-    group by received_at_date, activity_date, server_id, user_id
+    group by received_at_date, activity_date, server_id, user_id, client_type
```

Comment:
This extra column in the `group by` statement will result in multiple rows for the same columns as in the resulting primary key (`'received_at_date', 'activity_date', 'server_id', 'user_id'`), breaking incremental logic.

Reply:
Added `client_type` to the surrogate key

Reply:
Adding the `client_type` in the surrogate key will break the expectations for this model.